fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/workflow/CacheMode.java

```diff
@@ -0,0 +1,94 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.handlers.workflow;
+
+import com.yahoo.bard.webservice.config.BardFeatureFlag;
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+
+/**
+ * Caching config analyzer.
+ */
+public abstract class CacheMode {
```

Comment:
I think abstract class does not a whole lot of sense here. There are several options, 1. make it final class 2. make it a class and remove static modifier on getCacheMode to allow for override.